### PR TITLE
feat(toast): add global publisher

### PR DESCRIPTION
**Pull request by mamadoudicko**
# Description

This PR enhances the toast system by refactoring it and introduces the use of context to incorporate a global notification publisher.

## Checklist before requesting a review

Please delete options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented hard-to-understand areas
- [x] Any dependent changes have been merged

## Screenshots (if appropriate):
Screenshot 2023-05-27 at 12 06 50
Screenshot 2023-05-27 at 12 06 43
Screenshot 2023-05-27 at 12 06 28

### CODE REVIEW

**Review by StanGirard — APPROVED** [review]
Awesome job. A single use toast 😍

File: frontend/app/components/ui/Toast/hooks/useToastBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+import { ToastContent, ToastData, ToastPublisher } from "../domain/types";
+import { generateToastUniqueId } from "../helpers/generateToastUniqueId";
+
+// ⚠️ You should not probably import this. use `useToast` instead
+export const useToastBuilder = () => {
+  const [toasts, setToasts] = useState<ToastContent[]>([]);
+
```
Comment (iMADi-ARCH) [inline frontend/app/components/ui/Toast/hooks/useToastBuilder.ts:8]: Can you update `toggleToast` function to use the id of the toasts instead of index?

**Review by iMADi-ARCH — CHANGES_REQUESTED** [review]
Awesome PR!
I would have never thought of being so modular. I have a lot to learn from you. :)

File: frontend/app/components/ui/Toast/hooks/useToastBuilder.ts (hunk 1 above)
Comment (iMADi-ARCH) [reply frontend/app/components/ui/Toast/hooks/useToastBuilder.ts:8]: oops wrong code is showing 😅

File: frontend/app/components/ui/Toast/hooks/useToastBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    setToasts((toasts) =>
+      toasts.map((toast, i) => {
+        if (i === index) {
+          toast.open = value;
+        }
+        return toast;
+      })
+    );
+  };
+
```
Comment (iMADi-ARCH) [inline frontend/app/components/ui/Toast/hooks/useToastBuilder.ts:19]: Can you update `toggleToast` function to use the id of the toasts instead of index?

Comment (iMADi-ARCH) [inline frontend/app/components/ui/Toast/hooks/useToastBuilder.ts:19]: Can you update `toggleToast` function to use the id of the toasts instead of index?

Comment (mamadoudicko) [reply frontend/app/components/ui/Toast/hooks/useToastBuilder.ts:19]: Oow ! Thanks Man 😄

**Pull request merged**